src/TrustProxies.php

```diff
@@ -102,10 +105,24 @@ private function setTrustedProxyIpAddressesToTheCallingIp(Request $request)
     /**
      * Retrieve trusted header name(s), falling back to defaults if config not set.
      *
-     * @return array
+     * @return int A bit field of Request::HEADER_*, to set which headers to trust from your proxies.
      */
     protected function getTrustedHeaderNames()
     {
-        return $this->headers ?: $this->config->get('trustedproxy.headers');
+        $headers = $this->headers ?: $this->config->get('trustedproxy.headers');
```

Comment:
This sets the `$this->headers` property every time it's called. Perhaps check to see if `$this->headers` is set and simply return it? If the header hasn't been set, it can use configuration to determine the value and set it here, but only needs to do that once.

Reply:
The other caveat I see here is that Laravel users will never be able to configure this via environment unless the user explicitly removes the `protected $headers` property from the [TrustProxies middleware](https://github.com/laravel/laravel/blob/ddd8c2a7908a91e2efbbf916e2cd3806999a5357/app/Http/Middleware/TrustProxies.php#L22)

It might be worth calling that out in the Laravel documentation, as it's not super clear without source diving that since Laravel sets this headers property that environment cannot be used until you explicitly remove that value.

Reply:
@paulredmond that method is going to return a valid header always. Even if you do this in your Laravel middleware:
```php
protected $headers = 'INVALID HEADER';
```
Once approved I can submit a new PR to Laravel referencing these changes and including the config file and removing the `protected $headers` property.

Reply:
The net effect is the same, but this switch statement/check will run unnecessarily every time this is called, which @fideloper might not find to be a big deal, his call really.
 
The way I see it, the original intent of the `$headers` property was to allow configuration from an extended class, and if the property was null, it would use the configuration file. However, now you're going set `$this->headers` on each call even if that value comes from the existing `protected $headers` property.

My point is, why have the property if it's not used anywhere else in the class and this method now just overrides it each time anyway?